src/proj_types.jl

```diff
@@ -0,0 +1,50 @@
+## Types
+
+# Projection context.  TODO: Will this be exposed?
+#type Context
+#    rep::Ptr{Void} # Pointer to internal projCtx struct
+#end
+
+@doc """
+Cartographic projection type
+""" ->
+type Projection
+    #ctx::Context   # Projection context object
+    rep::Ptr{Void} # Pointer to internal projPJ struct
+    
+    # [geod]: a structure containing the parameters of the spheroid
+    # some of the fields in [geod] are mildly duplicative of the information
+    # available in [rep], which can be exposed only through _get_spheroid_defn
+
+    # It's needed as an argument for computing great circle distances though,
+    # and for most applications, users will only have to deal with a small number
+    # of projection objects, and so we precompute it for each Projection
+    geod::geod_geodesic
```

Comment:
It is unfortunate/wasteful (but not by much) in my mental model of a single projection for an entire dataset. Two alternatives here, neither particularly desirable to me:

1. Make the user construct their own `Geodesic` object from the `Projection`, before performing the geodesic/distance calculations.
2. Construct (and then discard) the `Geodesic` object everytime the user performs a geodesic/distance calculation.

The first one incurs a great mental object cost (knowing about projections and ellipses/etc, and how to transform between them, taking care to convert between xy and lonlat) for the end-user.

The second one is going to be alot more expensive for people who do need elliptical distance calculations (e.g. OpenstreetMap/etc).